src/selectors/verticalHeights.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * Copyright Schrodinger, LLC
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule verticalHeights
```

Comment:
Complicated logic for components of heights are here.  It's particularly tricky when dealing w/
maxHeight (the component will shrink to fit rows instead of putting footer after some padded spacing to fill height) and
ownerHeight (the component will fill a larger height allowing the page / owner to scroll, but the footer and scrollbar will always be visible within the owner height).

I added comments to explain things.  Please let me know if they're unclear or could be improved.